zio-sql/jvm/src/main/scala/Sql.scala

```diff
@@ -512,12 +512,36 @@ trait Sql {
 
     def desc: Ordering[Expr[F, A, B]] = Ordering.Desc(self)
 
-    def in[B1 >: B](set: Read[B1]): Expr[F, A, Boolean] = Expr.In(self, set)
+    def in[B1 >: B](set: Read[B1]): Expr[F, A, Boolean] = Expr.in(self, set)
 
     def widen[C](implicit ev: B <:< C): Expr[F, A, C] = {
       val _ = ev
       self.asInstanceOf[Expr[F, A, C]]
     }
+
+    def fold[Z](f: Expr.ExprCase[_, _, _, Z] => Z): Z =
+      caseValue match {
+        case c: Expr.AggregationCallCase[_, _, _, _, Expr[_, _, _]] =>
+          f(Expr.AggregationCallCase(c.param.fold(f), c.aggregation))
+        case c: Expr.BinaryCase[_, _, _, _, Expr[_, _, _]] =>
+          f(Expr.BinaryCase(c.left.fold(f), c.right.fold(f), c.op))
+        case c: Expr.FunctionCall1Case[_, _, _, _, Expr[_, _, _]] =>
+          f(Expr.FunctionCall1Case(c.param.fold(f), c.function))
+        case c: Expr.FunctionCall2Case[_, _, _, _, _, _, Expr[_, _, _]] =>
+          f(Expr.FunctionCall2Case(c.param1.fold(f), c.param2.fold(f), c.function))
+        case c: Expr.FunctionCall3Case[_, _, _, _, _, _, _, _, Expr[_, _, _] @unchecked] =>
```

Comment:
I get a warning here that the type is eliminated by erasure here but don't get any warnings with the variants with fewer type parameters so seemed like it is a compiler issue.